src/builtin.jq

```diff
@@ -258,8 +258,10 @@ def walk(f):
 # pathexps could be a stream of dot-paths
 def pick(pathexps):
   . as $in
-  | reduce path(pathexps) as $a (null;
-      setpath($a; $in|getpath($a)) );
+  | reduce path(pathexps) as $p (null;
+      ($p[] | select(. < 0 or type == "object")
```

Comment:
I'd rather not include this validation, or not here anyways, since this is not the only place we use path expressions.